### DISCUSSION TITLE
Why dosn't the droopdown menu update when I select a data set with radiobuttons?

### DISCUSSION THREAD

**Discussion by gcefalu** [quarto-dev/quarto-cli#14196] [Q&A]
### Description

I creating a Quarto presentation with revealjs format and shiny.  The radiobutton list displays data sets and the user selects one. When the data set is selected, a dropdown menu is updated with the variables of the new data set.  A second dropdown menu gets updated with the variables of the new data set.   The problem is that the dropdown menu is not being updated.  This code works very well in a shiny app, but it does not work in Quarto. Could you please help with this?

````qmd
---
title: "Linear Regression"
format: revealjs
editor: visual
profile: shiny
smaller: true
scrollable: true
---
```{r}
#| label: libraries
#| echo: true
suppressWarnings(suppressMessages(library(shinylive)))
suppressWarnings(suppressMessages(library(shiny)))
suppressMessages(suppressWarnings(library(rmarkdown)))
suppressMessages(suppressWarnings(library(dplyr)))
suppressMessages(suppressWarnings(library(bslib)))
```
```{r}
#| label: data
#| echo: true
#| message: true
#| warning: true

datasets <- list(Formaldehyde = Formaldehyde, Indometh = Indometh, 
                 LifeCycleSavings = LifeCycleSavings, OrchardSprays = OrchardSprays,                  Theoph = Theoph, chickwts = chickwts,faithful = faithful,                           randu = randu,ChickWeight = ChickWeight, Loblolly = Loblolly,                       anscombe = anscombe, beaver1 = beaver1, nottem = nottem,
                 stackloss = stackloss, swiss = swiss, DNase = DNase,
                 Nile = Nile, mtcars = mtcars, trees = trees)
```
```{r}
#| label: selectinput
#| echo: true
#| message: true
#| warning: true

  
  titlePanel(tags$h2("DATA SETS", style =    "color: blue;"))
  radioButtons("inDs", "Select dataset:", choices = names(datasets))
```
```{r}
#|context: server
ds <- reactive( datasets[[input$inDs]] )
```
```{r}
varSelectInput("inVar", "Select variable:", datasets[[1]])
varSelectInput("inVar1", "Select a second variable:", datasets[[1]])
```
```{r}
#|context: server
observe(updateVarSelectInput(session, "inVar", data = ds()))
observe(updateVarSelectInput(session, "inVar1", data = ds()))
```

**Comment by mcanouil** [comment]
@gcefalu Could you edit the title?
Could you also ensure your example is valid, I can see ` ```{r}#| label: data` which is clearly not valid code.

Finally, I highly suggest you take a look at the documentation:

- https://quarto.org/docs/interactive/shiny/

**Comment by gcefalu** [comment]
Thanks again. I will l. Sent from my iPhoneOn Mar 11, 2026, at 3:37 PM, Mickaël Canouil ***@***.***> wrote:﻿
@gcefalu Could you edit the title?
Could you also ensure your example is valid, I can see  ```{r}#| label: data which is clearly not valid code.
Finally, I highly suggest you take a look at the documentation:

https://quarto.org/docs/interactive/shiny/

**Comment by gcefalu** [comment]
Thank you for adding the quarto-cli folder to my git-hub repository account.
________________________________

**Comment by gcefalu** [comment]
Should I copy the folder to Program Files in Windows?
________________________________
